tools/odbc/statement.cpp

```diff
@@ -144,7 +161,102 @@ SQLRETURN SQLColumns(SQLHSTMT statement_handle, SQLCHAR *catalog_name, SQLSMALLI
 SQLRETURN SQLColAttribute(SQLHSTMT statement_handle, SQLUSMALLINT column_number, SQLUSMALLINT field_identifier,
                           SQLPOINTER character_attribute_ptr, SQLSMALLINT buffer_length, SQLSMALLINT *string_length_ptr,
                           SQLLEN *numeric_attribute_ptr) {
-	throw std::runtime_error("SQLColAttribute"); // TODO
+
+	return duckdb::WithStatementPrepared(statement_handle, [&](duckdb::OdbcHandleStmt *stmt) {
+		if (column_number < 1 || column_number > stmt->stmt->GetTypes().size()) {
+			stmt->error_messages.emplace_back("Column number out of range.");
+			return SQL_ERROR;
+		}
+
+		duckdb::idx_t col_idx = column_number - 1;
+
+		switch (field_identifier) {
+		case SQL_DESC_LABEL: {
+			if (buffer_length <= 0) {
+				stmt->error_messages.emplace_back("Inadequate buffer length.");
+				return SQL_ERROR;
+			}
+
+			auto col_name = stmt->stmt->GetNames()[col_idx];
+			auto out_len = snprintf((char *)character_attribute_ptr, buffer_length, "%s", col_name.c_str());
+			if (string_length_ptr) {
+				*string_length_ptr = out_len;
+			}
+
+			return SQL_SUCCESS;
+		}
+		case SQL_DESC_OCTET_LENGTH:
+			// 0 DuckDB doesn't provide octet length
+			*numeric_attribute_ptr = 0;
```

Comment:
Should we check if `numeric_attribute_ptr` is nullptr?

Reply:
Good catch, I will add that check.